api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -31,6 +31,7 @@
 	private Integer minLength; // If > 0, will always return identifiers with a minimum of this length
 	private Integer maxLength; // If > 0, will always return identifiers no longer than this length
     private String baseCharacterSet; // Enables configuration in appropriate Base
+	private Integer nextSequenceValue; // Pre-generate the next identifier
 
```

Comment:
Although I don't think having a publicly available mutation of state as this is a good idea, we do need this for cases where the Identifiers are assigned offline in remote hospitals and then data is entered later in Bahmni. However, we never read this property to assign identifiers (which is a suboptimal design and I agree that we should change this).

I double checked the synchronised block here - https://github.com/openmrs/openmrs-module-idgen/blob/master/api/src/main/java/org/openmrs/module/idgen/processor/SequentialIdentifierGeneratorProcessor.java#L44 as well as the IdentifierSourceServiceLoadTest that @djazayeri mentioned in IDGEN-6 (https://issues.openmrs.org/browse/IDGEN-6) and this seems not to be a breaking change. Without this change, we might have to rework our support for offline identifier distribution. 

Our use case specifically - read nextSequenceValue to check if the offline Id assigned is next in line, prompt the registration admin if this is not the case to force update the nextSequenceValue. In all other use cases, we do not use the nextSequenceValue.

(Per @mogoodrich 's suggestion, I may spike a post on /ws/rest/v1/identifierSource/<uuid>/identifier (or similar) with the post data containing the nextSequenceValue - this new value gets validated and then the state of the SequentialIdentifierGenerator is updated, the REST client then needs to make another call to get the identifier as in the regular case).